### PR TITLE
Integrated the report functionality into the main and viewStudent UI,…

### PR DESCRIPTION
**Pull request by J8cob1**
… and did some fixing/refactoring

### CODE REVIEW

File: report.py [hunk 1]
```diff
@@ -84,30 +61,30 @@ def get(self):
             is_final = True
 
         try:
-            pdf = _make_printable_reports(session_id, team_id, is_final)
+            pdf = _make_printable_reports(session_id, is_final)
```
Comment (J8cob1) [inline report.py:64]: Maybe we don't need to call this "pdf" anymore?

**Comment by J8cob1** [comment]
Is your task complete?
Yes? Adam's UI changes have been incoporated into the existing user interface, though in a slightly modified way. The old way of accessing the data (/reportList) has been removed.

Is the branch caught up to master/are there merge conflicts still to be resolved?
Caught up to master

Are old files unnecessarily introduced?
Nope. Only changing existing files.

Is old code unnecessarily overwritten?
I would think no, mostly. There may be some times where I do some stuff that is not entirely necessary (I added a newline in one file, and added some try/except and custom exceptions that I am not sure if we need), but overall hopefully not that much?

Are new issues introduced to existing code? Do existing automated tests continue to work? [to run]
Automated tests: pass
Log in: pass (with existing token)
Dashboard/Edit: pass
Set Dates: pass
ReportList: removed in this change
View Student: pass
View Review: pass
Anonymized (Student) Report Generation for Every Student in a Session: pass
Unanonymized (Professor) Report Generation for an Individual Student: pass

Has flake8 been checked? - Automagically, yes

uses 4 spaces where appropriate, not tabs?
Yes

camelCase for class names, snake_case for variable and function names?
Yes

code is commented?
I don't think I added any new comments

Does the code handle edge cases? Are try/except statements used sufficiently?
I added some try/except statements. I am not sure if I handled all the edge cases.

**How to test:**
**To test generation of anonymous reports:**
1. Fill out some reviews. You could do this using the /reviews page or by creating some reviews by executing the sql queries at the bottom of this comment (executing the queries won't be the same as filling out reviews--there might be certain attributes in the students table that don't get updated--but it gives you some data to work with)
2. In the professor dashboard, select the session with the review data you just added and click the "Generate Reports" dropdown and click "Midterm" 
3. Verify that anonymous midterm reports are generated for every student in the currently selected session. The students in the report should be ordered by team. If you used the sql queries to get review data, The members of team 0 in session 0 will have review data. The other teams in that session (and in every other session) won't. 
4. Go back to the previous page using the back button
5. Verify that the "Generate Reports" dropdown still has "Select Report Type" as the currently selected value (it is supposed to reset back to that every time you select "Midterm" or "Final")
6. Then click the "Generate Reports" dropdown again and click "Final"
7. Verify that anonymous final reports are generated for every student in the currently selected session, with the students being organized by team in the same way as it was for the midterm report

**To test unorganized professor report generation:**
1. Generate or fill out some reports (see step 1 of the anonymous report generation process described above) if you still need some
2. In the professor dashboard, select the session that has the review data and click the name of a student (that has review data--the score ranges should indicate which ones do) to be taken to the view student page for that student
3. Click the "Report" text at the bottom of the "Midterm" box
4. Verify that an unanonymized midterm report has been generated for that student
5. Go back to the view student page and click the "Report" text at the bottom of the "Final" box
6. Verify that an unanonymized final report has been generated for that student (having a midterm report or two or more might help you make sure it is generated properly) 

**Open up the capstone360.db file in sqlite3 and paste these commands in to get some midterm and final reports:**
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,0,0,0,1,2,3,4,5,4,3,2,1,2,0,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,0,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,0,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,40,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,0,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,0,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,0,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,10,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,0,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,0,0,0,1,2,3,4,5,4,3,2,1,2,0,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,0,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,0,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,0,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,0,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,0,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,0,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,0,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,0,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,0,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,0,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,0,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,100,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,6,0,0,1,2,3,4,5,4,3,2,1,2,20,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,6,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,0,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,6,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,6,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,20,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,6,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,6,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,6,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,20,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,6,0,0,1,2,3,4,5,4,3,2,1,2,20,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,6,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,0,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,6,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,6,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,6,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,30,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,6,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,6,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,12,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,12,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,12,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,0,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,12,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,12,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,40,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,12,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,12,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,12,0,0,1,2,3,4,5,4,3,2,1,2,20,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,12,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,12,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,0,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,12,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,30,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,12,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,12,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,12,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,18,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,18,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,20,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,18,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,18,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,0,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,18,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,18,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,18,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,30,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,18,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,18,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,18,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,18,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,0,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,18,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,30,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,18,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,30,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,18,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,24,0,0,1,2,3,4,5,4,3,2,1,2,20,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,24,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,20,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,24,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,20,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,24,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,20,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,24,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,0,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,24,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,10,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,24,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,10,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,24,0,0,1,2,3,4,5,4,3,2,1,2,15,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,24,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,15,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,24,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,25,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,24,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,15,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,24,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,0,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,24,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,25,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,24,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,15,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,30,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,30,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,30,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,30,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,30,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,30,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,0,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,30,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,50,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,30,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,30,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,30,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,10,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,30,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,30,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,10,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,30,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,0,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,30,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,50,'thth','rgerg','egweg',NULL,NULL,1,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,36,0,0,1,2,3,4,5,4,3,2,1,2,20,'ehehe','grggr','ergg','ethheth',NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,36,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,10,'eheth','rgerg','gwegwe',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,36,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,20,'ethte','gwrgre','gwegw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,36,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,36,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,20,'rheherh','ewgw','rgrwgw',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,36,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,20,'erher','herher','rgr',NULL,NULL,0,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,36,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,0,'thth','rgerg','egweg',NULL,NULL,0,0);

INSERT INTO "reports" VALUES('2019-04-17 22:11:17.535073',0,36,0,0,1,2,3,4,5,4,3,2,1,2,10,'ehehe','grggr','ergg','ethheth',NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.540400',0,36,0,6,2,2,2,2,2,2,2,NULL,NULL,NULL,20,'eheth','rgerg','gwegwe',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.543028',0,36,0,12,3,3,3,3,3,3,3,NULL,NULL,NULL,30,'ethte','gwrgre','gwegw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.550765',0,36,0,18,4,4,4,4,4,4,4,NULL,NULL,NULL,10,'hrhed','ewf','efew',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.556732',0,36,0,24,5,5,5,5,5,5,5,NULL,NULL,NULL,15,'rheherh','ewgw','rgrwgw',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.568274',0,36,0,30,4,3,4,3,4,3,4,NULL,NULL,NULL,15,'erher','herher','rgr',NULL,NULL,1,0);
INSERT INTO "reports" VALUES('2019-04-17 22:11:17.570883',0,36,0,36,1,2,1,2,1,2,1,NULL,NULL,NULL,0,'thth','rgerg','egweg',NULL,NULL,1,0);

**Review by pdx-cs-lbryan — CHANGES_REQUESTED** [review]
Is your task complete?
-builds on reports generation

Is the branch caught up to master/are there merge conflicts still to be resolved?
ahead of master

Are old files unnecessarily introduced?
No

Is old code unnecessarily overwritten?
no

Are new issues introduced to existing code? Do existing automated tests continue to work?

Log in: yes
Dashboard: yes
shows score range? looked to be working
Edit: yes
Review: yes
Prof View of Student Reports: yes
Report Generation session: yes
Report Generation individual: yes
Set/Add Date: yes
Automated Tests: pass

Uses 4 spaces where appropriate, not tabs?
yes

camelCase for class names, snake_case for variable and function names?
yes

code is commented?
**some methods need comments**

Does the code handle edge cases? Are try/except statements used sufficiently?
**some methods need error handling**

Is there documentation on how to test your code somewhere?
extensive

File: gbmodel/model_sqlalchemy.py [hunk 2]
```diff
@@ -122,12 +122,14 @@ def dashboard(self, session_id):
         sessions = session.get_sessions()
         return lists, sessions
 
-    def get_team_name_from_id(self, team_id):
-        team_name_obj = teams.query.filter_by(id=team_id).first()
-        if team_name_obj is not None:
-            return team_name_obj.name
-        else:
-            return None
+    def get_team_from_id(self, team_id):
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: this method needs comments

File: gbmodel/model_sqlalchemy.py [hunk 3]
```diff
@@ -122,12 +122,14 @@ def dashboard(self, session_id):
...
+        """
+        Get the team object associated with the given id
+        Input: team_id
+        Output: a team object, if found. None otherwise
+        """
+        result = teams.query.filter(teams.id == team_id).first()
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: no error handling

File: gbmodel/model_sqlalchemy.py [hunk 4]
```diff
@@ -182,6 +184,26 @@ def get_team_members(self, tid):
             return None
         return mems
 
+    def get_students_in_session(self, session_id):
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py:190]: this method needs error handling

File: gbmodel/model_sqlalchemy.py [hunk 5]
```diff
@@ -182,6 +184,26 @@ def get_team_members(self, tid):
...
+        results = students.query.filter(students.session_id == session_id).order_by(students.tid.asc()).all()
+        return results
+
+    def get_student_in_session(self, sid, session_id):
+        """
+        Get a student from the students table
+        Input: student id, session id
+        Output: the student that we found, or none if nothing was found
+        """
+        result = students.query.filter(students.id == sid, students.session_id == session_id).first()
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: no error handling

File: gbmodel/model_sqlalchemy.py [hunk 6]
```diff
@@ -414,17 +436,17 @@ def check_review_state(self, session_id, date):
 class reports(db.Model):
     __table__ = db.Model.metadata.tables['reports']
 
-    def get_reports_for_student(self, student_id, term_id, is_final=None):
+    def get_reports_for_student(self, student_id, session_id, is_final=None):
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py:467]: needs comments on input/ouput

File: report.py [hunk 7]
```diff
@@ -13,33 +13,11 @@ class MissingStudentException(Exception):
...
-        if len(teams) <= 0:
-            return make_response(render_template('404.html'), 404)
-
-        return render_template('reportList.html',
-                               students=students,
-                               sessions=sessions,
-                               teams=teams,
-                               team_id=team_id,
-                               session_id=0)
+class MissingTeamException(Exception):
```
Comment (pdx-cs-lbryan) [inline report.py:16]: should be 'missingTeamException

Comment (J8cob1) [reply report.py:16]: Are you sure? That's not how the existing class is written (nor how any of our classes are written?)

File: gbmodel/model_sqlalchemy.py (hunk 4 above)
Comment (J8cob1) [reply gbmodel/model_sqlalchemy.py:190]: Done

File: gbmodel/model_sqlalchemy.py [hunk 8]
```diff
@@ -182,6 +187,33 @@ def get_team_members(self, tid):
...
+        """
+        Gets a list of students in the given session, ordered by team (in ascending order)
+        Input: session_id
+        Output: the list of students
+        """
+        # https://stackoverflow.com/questions/4186062/sqlalchemy-order-by-descending
+        # https://docs.sqlalchemy.org/en/13/orm/query.html
+        try:
+            results = students.query.filter(
+                          students.session_id == session_id).order_by(students.tid.asc()).all()
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:200]: If I may ask, what kind of indentation strategy do we want here? Does what I did here work?

Comment (pdx-cs-lbryan) [reply gbmodel/model_sqlalchemy.py:200]: I think that's fine

File: gbmodel/model_sqlalchemy.py [hunk 9]
```diff
@@ -432,18 +464,26 @@ def check_review_state(self, session_id, date):
 class reports(db.Model):
     __table__ = db.Model.metadata.tables['reports']
 
-    def get_reports_for_student(self, student_id, term_id, is_final=None):
+    def get_reports_for_student(self, student_id, session_id, is_final=None):
         """
-        Gets all available reports for a student, optionally filtering to only midterms or finals.
+        Gets all available reports for a student, optionally filtering to only midterms or finals
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:469]: I wonder if we might consider changing the word "reports" to "reviews" here. If so, it might be wise to go through the entire code base and do this kind of thing

Comment (pdx-cs-lbryan) [reply gbmodel/model_sqlalchemy.py:469]: It'd be a good thing for code clean up, I don't think we want to try doing that now

**Comment by pdx-cs-lbryan** [comment]
Automated tests: pass
Log in: pass
Dashboard/Edit: pass
Set Dates: pass
View Student: pass
View Review: pass
Anonymous report/session report generation: pass
Non-anonymous report/individual report generation: pass
Edit student: pass

**Review by pdx-cs-lbryan — APPROVED** [review]
